### PR TITLE
[FIX]l10n_pa_localization: fixed onchange methods on resparter for address fields, Use New api instead of Old api

### PR DESCRIPTION
**Pull request by hugho-ad**
https://github.com/Vauxoo/yoytec/pull/849

### CODE REVIEW

**Comment by hugho-ad** [comment]
@luistorresm can you review please

File: l10n_pa_localization/model/res_partner.py [hunk 1]
```diff
@@ -238,61 +230,45 @@ def _display_address(
...
-                'hood_id': hood_id.id,
-                'township_id': hood_id.township_id.id or par.township_id.id,
-                'district_id': hood_id.district_id.id or par.district_id.id,
-                'state_id': hood_id.state_id.id or par.state_id.id,
-                'country_id': hood_id.country_id.id or par. country_id.id
-            }
-        self.write(value)
-        return {'value': value}
+            par.township_id = par.hood_id.township_id or par.township_id
+            par.district_id = par.hood_id.district_id or par.district_id
```
Comment (luistorresm) [inline l10n_pa_localization/model/res_partner.py]: @hugho-ad 

If only you set the township_id, the change in that field will call the your onchange.

You not need assign all the values in this method

Can you try this?

File: l10n_pa_localization/model/res_partner.py [hunk 2]
```diff
@@ -238,61 +230,45 @@ def _display_address(
...
+            if par.district_id.state_id != par.state_id:
+                par.district_id = None
+            par.country_id = par.state_id.country_id or par.country_id
+
+    @api.onchange("country_id")
+    @api.multi
+    def _onchange_country_id(self):
+        for par in self:
+            if par.state_id.country_id != par.country_id:
+                par.sate_id = None
```
Comment (luistorresm) [inline l10n_pa_localization/model/res_partner.py]: par.sate_id must be par.state_id

**Comment by hugho-ad** [comment]
@luistorresm thanks!

@nhomar can you merge this please

File: l10n_pa_localization/model/res_partner.py [hunk 3]
```diff
@@ -238,61 +230,45 @@ def _display_address(
...
-                'hood_id': hood_id.id,
-                'township_id': hood_id.township_id.id or par.township_id.id,
-                'district_id': hood_id.district_id.id or par.district_id.id,
-                'state_id': hood_id.state_id.id or par.state_id.id,
-                'country_id': hood_id.country_id.id or par. country_id.id
-            }
-        self.write(value)
-        return {'value': value}
+            par.township_id = par.hood_id.township_id or par.township_id
+            # par.district_id = par.hood_id.district_id or par.district_id
```
Comment (luistorresm) [inline l10n_pa_localization/model/res_partner.py]: @hugho-ad 

Commented lines

**Comment by hugho-ad** [comment]
@nhomar can you merge this please

**Pull request merged**